katportalclient/test/test_client.py

```diff
@@ -675,25 +673,24 @@ def test_sensor_detail(self):
         history_base_url = self._portal_client.sitemap[
             'historic_sensor_values']
         sensor_name = 'anc_weather_wind_speed'
-
         self.mock_http_async_client().fetch.side_effect = mock_async_fetcher(
-            valid_response='[{}]'.format(
-                sensor_json['anc_weather_wind_speed']),
-            invalid_response='[]',
+            valid_response=('{"data":[%s]}' % sensor_json['anc_weather_wind_speed']),
+            invalid_response='{"error":"test error"}',
             starts_with=history_base_url,
             contains=sensor_name)
 
         with self.assertRaises(SensorNotFoundError):
             yield self._portal_client.sensor_detail('invalid_sensor_name')
 
         sensor_detail = yield self._portal_client.sensor_detail(sensor_name)
-        self.assertTrue(sensor_detail['name'] == sensor_name)
-        self.assertTrue(sensor_detail['description'] == "Wind speed")
-        self.assertTrue(sensor_detail['params'] == "[0.0, 70.0]")
-        self.assertTrue(sensor_detail['units'] == "m/s")
-        self.assertTrue(sensor_detail['type'] == "float")
-        self.assertTrue(sensor_detail['component'] == "anc")
-        self.assertTrue(sensor_detail['katcp_name']
+        
+        self.assertTrue(sensor_detail[0] == sensor_name)
+        self.assertTrue(sensor_detail[2]['description'] == "Wind speed")
```

Comment:
Having to dig into arbitrary indexes to get at the useful fields seems harder than necessary to use.  Can't we just repackage the katstore reply into the old format so this test doesn't need to change?  The docstring for `sensor_detail` still promises a dict:  https://github.com/ska-sa/katportalclient/blob/068331fa84b27f8136dc5e463d2b1b5bd70e071a/katportalclient/client.py#L1100-L1113

Reply:
+1

Reply:
@ajoubertza this is addressed in incoming commit. 
new katstore does not have all the sensor info in its sensors_meta table.
example of output below.
```
{"type": "boolean", "component": "mon_proxy", "description": "Whether monitor is currently connected to cbfmon_1", "original_name": "mon_proxy.cbfmon_1-connected"} 
```

Reply:
Thanks.  Test needs to be reverted then.
And as mentioned in previous comment, we need the units.